extensions/generate-index-data.js

```diff
@@ -0,0 +1,192 @@
+      return;
+    }
+
+    try {
+      // Process each defined index set
+      for (const [setName, setParams] of Object.entries(setsConfig)) {
+        logger.info(`Processing index set: ${setName}`);
+        const { component, family, filter, env_type, version, output_file, attribute_name } = setParams;
+
+        // Validate required parameters
+        const missingParams = [];
+
+        if (!component) missingParams.push('component');
+        if (!family) missingParams.push('family');
+        if (!attribute_name) missingParams.push('attribute_name');
+
+        if (missingParams.length > 0) {
+          logger.warn(`Missing required parameter(s) (${missingParams.join(', ')}) for set "${setName}". Skipping.`);
+          continue;
+        }
+
+        // Determine the target version
+        let targetVersion = version || null;
+
+        if (version === 'latest') {
+          // Retrieve the component to access 'latest'
+          const componentObj = contentCatalog.getComponents().find(comp => comp.name === component);
+          if (componentObj && componentObj.latest && componentObj.latest.version) {
+            targetVersion = componentObj.latest.version;
+            logger.debug(`Resolved 'latest' to version "${targetVersion}" for component "${component}".`);
+          } else {
+            targetVersion = null; // Represents unversioned
+            logger.info(`Component "${component}" is unversioned. Using the sole available version.`);
+          }
+        }
+
+        // Gather items based on the target version
+        const items = gatherItems(contentCatalog, {
+          component,
+          family,
+          filter,
+          env_type,
+          version: targetVersion
+        }, logger);
+
+        if (!items.length) {
+          logger.warn(`No items found for set "${setName}". Skipping output.`);
+          continue;
+        }
+
+        const uniqueItems = deduplicateItems(items);
+        await addDataAttributeToComponents(uniqueItems, contentCatalog, attribute_name, logger);
+        if (output_file) createIndexFile(uniqueItems, siteCatalog, output_file, logger);
+      }
+    } catch (error) {
+      logger.error(`Failed to generate indexes: ${error.stack}`);
+    }
+  });
+};
+
+/**
+ * Gathers items (pages) from the contentCatalog based on the provided setParams.
+ * setParams should contain:
+ *   - component (string): Component name to search
+ *   - family (string): Family of pages (usually 'page')
+ *   - version (string|null): The component version ('latest', specific version, or null for unversioned)
+ *   - filter (string, optional): A substring to match in the URL or other criteria
+ *   - env_type (string, optional): Deployment type to match ('Docker', 'Kubernetes', 'Linux', 'Redpanda Cloud')
+ */
+function gatherItems(contentCatalog, setParams, logger) {
+  const { component, family, filter, env_type, version } = setParams;
+  let pages = [];
+
+  if (version) {
```

Comment:
This feels like code duplication but I can't say for sure. If you don't agree, disregard.